### ISSUE TITLE
Add clear option to the select editor when used a a header filter

### ISSUE THREAD

**Issue by qlj** [olifolkerd/tabulator#3293]
Hi Oli,

Is there already a way to improve a column select header. like this:

`columns: [ {title:"Title", field:"title", headerFilter:"select", headerFilterParams: {values:true} },`

so that it populates with values hosted in the column PLUS 'All', that the user can select and which clears filtering on the column.

![Screen Shot 04-30-21 at 09 16 PM](https://user-images.githubusercontent.com/104757/116750309-299ceb80-a9fa-11eb-91fd-31b6fdeae415.PNG)

**Comment by purxiz** [comment]
For the time being, from the docs you could do something like this, though it's not a very elegant solution:

```
{title:"Name", field:"name", editor:"select", editorParams:function(cell){
    //create a options list of all values from another column in the table
    var rows = table.getRows();
    var values = {};

    rows.forEach(function(row){
        var data = row.getData();

        values[data.fullname] = data.fullname;
    });
    values[''] = 'All';


    return {values:values};
}
```

**Comment by hhoeflin** [comment]
would be cool feature to have. Is very confusing not to be able to deselect the filter.

**Comment by olifolkerd** [comment]
I would not want an all option as the wording for this is very context specific and not something that the formatter should decide, pushing the empty value to the top of the values list is the correct approach, though i could certainly look to add a clear cross to it like the autocomplete editor has.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
I have just pushed updates to the 5.2 branch that replaces the ***select*** and ***autocomplete*** editors with the new ***list*** editor, which will add this feature.

It will be included in the 5.2 release later this month.

Cheers

Oli :)
